### PR TITLE
importccl: create tables before ingesting data

### PR DESCRIPTION
**Pull request by dt**
importccl: create tables before ingesting in IMPORT

This changes the order in which IMPORT creates tables and ingests data.

Previously, IMPORT computed the tables it would create, but kept them only in the job record during ingestion.
Only after ingestion completed successfully, in the successful completion hook of the IMPORT job, did it then write those table descriptors and create namespace mappings.

This change switches IMPORT to create the tables in a non-public INGESTING state first, then run the ingestion job, and then simply move them from ingesting to public when it finishes.

This has some advantages and disadvantages. Creating the descriptors early has some nice side-effects: it avoids a potential race where another table could be created, imported or restored with a conlficting name during ingestion, and things try to map KVs to tables can actually see the table, so special cases like enforced splits between tables should work as expected during ingestion, fixing the issue where an importing table could be merged into another table after the ingestion job (which suspends merges) finished but before the descriptors were written. Additionally the admin UI or other places that surface tables can show them if they so choose, which could, e.g. help explain where disk-space is being used by not-yet-public tables.

Unfortunately it also introduces some disadvantages, most notably that writing the descriptors adds more ways by which traffic might end up in the ingesting range and thus more code paths that need ot be audited and must include the proper checks for non-public states, plus writing any incomplete state before the job is finished adds more that needs to be properly cleaned up on failure.

Overall though, importing into an existing table will need to solve the same downsides in any case, so making imports of new tables behave the same as imports into existing tables -- by importing into a just-created existing table -- will simplify the overall code.

Release note: none.

There's one special-case where IMPORT _never_ creates tables, when running with the `transform` option that writes a BACKUP instead. This mode was easily to maintain in the initial version of IMPORT which internally essentially made a BACKUP anyway and then restored it. As IMPORT has diverged from that, and become more focused on directly ingesting data, maintaining this functionality has become more expensive. Given that simply backing up after ingesting can generate a the same backup, maintaining this extra overhead in the IMPORT code no longer makes sense. 

Release note (sql change): remove 'transform' option from IMPORT.

### CODE REVIEW

**Comment by dt** [comment]
I split the `workload` part of this into its own change over in #37343 and I'll rebase this one on that after it merges.

File: pkg/ccl/importccl/import_stmt.go
Comment (mjibson) [inline pkg/ccl/importccl/import_stmt.go]: is this leftover debug stuff?

File: pkg/ccl/importccl/import_stmt.go
Comment (mjibson) [inline pkg/ccl/importccl/import_stmt.go]: how hard is this? seems kinda important.

File: pkg/ccl/importccl/import_stmt.go
Comment (mjibson) [inline pkg/ccl/importccl/import_stmt.go]: can `BackupPath` be removed from the .proto?

File: pkg/ccl/importccl/import_stmt.go
Comment (mjibson) [inline pkg/ccl/importccl/import_stmt.go]: So this else case is for that not-yet-written import into existing tables feature?

File: pkg/ccl/importccl/import_stmt.go
Comment (dt) [reply pkg/ccl/importccl/import_stmt.go]: agreed it is important. At first glance, it'll be pretty tricky and will require refactoring the jobs code a bit. I looked at it for a bit but decided that it was enough going to involve enough moving things around that I'd rather not do it here and leave it to a self-contained change later, but I'd certainly consider something that would need to be addressed before release.

File: pkg/ccl/importccl/import_stmt.go
Comment (dt) [reply pkg/ccl/importccl/import_stmt.go]: Eh, I realized that for now I'd like to keep it and assert it is empty to avoid adopting a `transform` job on a node that doesn't support `transform`

File: pkg/ccl/importccl/import_stmt.go
Comment (dt) [reply pkg/ccl/importccl/import_stmt.go]: Correct

**Review by madelynnblue — APPROVED** [review]

**Pull request merged**
